### PR TITLE
[Lib][Nfs] Add support for windows clients

### PR DESCRIPTION
**Pull request by pranavprakash20**
# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Comment by pranavprakash20** [comment]
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-E390Y8/

**Comment by pranavprakash20** [comment]
log: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-TUZT5N/

**Review by MohitBis — APPROVED** [review]
LGTM

File: tests/nfs/nfs_verify_windows_client.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+            fs_name,
+            nfs_export,
+            fs,
+            ha,
+            vip,
+            ceph_cluster=ceph_cluster,
+        )
+        # If windows clients are present, perform mount
+        if win_clients:
+            cmd = f"mount -o {nfs_nodes[0].ip_address}:/export_1 Z"
```
Comment (msainiRedhat) [inline tests/nfs/nfs_verify_windows_client.py:76]: Need to revalidate the shared mount unmount command for windows.

File: suites/reef/nfs/tier1-nfs-ganesha-windows-clients.yaml [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+              args:
+                - "rbd pool init rbd"
+              command: shell
+      desc: bootstrap and deploy services.
+      destroy-cluster: false
+      polarion-id: CEPH-83573713
+      module: test_cephadm.py
+      name: Deploy cluster using cephadm
+
+  - test:
```
Comment (msainiRedhat) [inline suites/reef/nfs/tier1-nfs-ganesha-windows-clients.yaml:70]: 4 Linux client required? We are validating the windows client. Only 1 linux client will be needed to change the permission for windows share.

**Review by msainiRedhat — CHANGES_REQUESTED** [review]

File: suites/reef/nfs/tier1-nfs-ganesha-windows-clients.yaml (hunk 2 above)
Comment (pranavprakash20) [reply suites/reef/nfs/tier1-nfs-ganesha-windows-clients.yaml:70]: Noted, removing 3 linux clients

File: tests/nfs/nfs_verify_windows_client.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    """Verify NFS HA cluster creation
+    Args:
+        **kw: Key/value pairs of configuration information to be used in the test.
+    """
+    config = kw.get("config")
+    nfs_nodes = ceph_cluster.get_nodes("nfs")
+    clients = ceph_cluster.get_nodes("client")
+
+    port = config.get("port", "2049")
+    version = config.get("nfs_version", "4.2")
```
[the remark this answers is not in the document]
Comment (pranavprakash20) [reply tests/nfs/nfs_verify_windows_client.py:23]: Noted, making 3 as default

**Comment by pranavprakash20** [comment]
New run log with moutn and umount https://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-HVYJP1/

**Review by msainiRedhat — APPROVED** [review]
LGTM

**Pull request merged**